boards/build_scripts/tock_kernel_layout.ld

```diff
@@ -109,6 +91,23 @@ SECTIONS
         KEEP(*(.riscv.trap_vectored));
         KEEP(*(.riscv.trap));
 
+        /* x86     
+         * We need to make sure that qemu-system-i386 finds the multiboot header within the
+         * first 8K of the ELF. The multiboot header is placed at the beginning of the .text section, so
+         * this section has to be placed first in the file.
+         *
+         * The multiboot header's documentation is available at
+         * https://www.gnu.org/software/grub/manual/multiboot/multiboot.html.
+         * This allows Tock to be booted using a bootloader that supports multiboot v1.
+         * QEMU provides for x86 such a bootloader when used with the `-kernel` flag.
+         *
+         * The bootloader is responsable for loading the kernel into RAM and executing it. Without it,
+         * the Tock kernel should provide a bootloader that starts reading the kernel into RAM.
+         *
+         * This is the reason why the .text section has to be the first.
+         */
+        KEEP(*(.x86.start))
```

Comment:
I think we should just update [the board file to explicitly define `.multiboot` section](https://github.com/tock/tock/blob/master/boards/qemu_i486_q35/src/main.rs#L44), which we can then explicitly place in the linker script.

This doesn't really have anything to do with x86, it just happens to be that only our x86 platforms currently use multiboot.

Reply:
Done.